### DISCUSSION TITLE
cgns for octree grids

### DISCUSSION THREAD

**Discussion by AndreasLieta** [CGNS/CGNS#340] [CGNStalk]
I need to solve three problems. First, I need to create a grid based on octrees. I use several refinement criteria, but all are boolean functions of coordinates. To divide a cell or not. And, similarly, a boolean function of coordinates that determines whether the cell is in the computational area. It is for deleting. Secondly, I have to put CFD data into the grid to each terminal cell and to each terminal face. And, thirdly, I have to perform a parallel loading of the grid with data on a multiprocessor MPI cluster. In this case, each processor must receive a list of adjacent processors, and each pair of processors with adjacent partitions must receive the same list of interface terminal faces in order to build the same buffers.

I already have code that solves these problems in my own format. Can I use cgns format and suggested libraries for this? What confuses me is the example when base grid has two cells, right base cell is divided into eight and the four rightmost small cells are further divided. Then we have two levels of one-to-many relationships. I'm afraid of hanging nodes and faces here. Will the parallel code be able to restore the relationship between the cells and faces? I need each cell to know its faces, and each face to know its pair of adjoining cells.

Thanks in advance!

**Comment by MicK7** [comment]
Hi not sure if I understood your point correctly. It seems that the CGNS storage of NGON and NFACE could help with your cell-face relationship.

**Reply by AndreasLieta** [reply]
Thanks for the advice! I will try to use these elements
